pkg/osvscanner/filter.go

```diff
@@ -12,22 +12,29 @@ import (
 	"github.com/ossf/osv-schema/bindings/go/osvschema"
 )
 
-// filterUnscannablePackages removes packages that don't have enough information to be scanned
+// filterUnscannablePackages removes packages that don't have enough information to be scanned or
+// are not a supported ecosystem
 // e,g, local packages that specified by path
 func filterUnscannablePackages(scanResults *results.ScanResults) {
 	packageResults := make([]imodels.PackageScanResult, 0, len(scanResults.PackageScanResults))
 	for _, psr := range scanResults.PackageScanResults {
 		p := psr.PackageInfo
 
 		switch {
-		// If none of the cases match, skip this package since it's not scannable
+		// If **none** of the cases match, skip this package since it's not scannable
 		case !p.Ecosystem().IsEmpty() && p.Name() != "" && p.Version() != "":
 		case p.Commit() != "":
-		case p.Ecosystem().Ecosystem == osvschema.EcosystemMaven && p.Name() == "unknown":
 		default:
 			continue
 		}
 
+		switch {
+		// If **any** of the following cases are true, skip this package
+		case p.Ecosystem().Ecosystem == osvschema.EcosystemMaven && p.Name() == "unknown",
+			!p.Ecosystem().IsValid() && !p.Ecosystem().IsEmpty():
+			continue
```

Comment:
Wasn't before `p.Ecosystem().Ecosystem == osvschema.EcosystemMaven && p.Name() == "unknown"` would **not** continue (since there's no implicit `fallthrough` in go), and now after it is continuing?

Am I interpreting this right?

Reply:
yes, this is a behavior change, I think the previous behavior is a bug.